### DISCUSSION TITLE
Create custom event to publish to the Event-Bus.

### DISCUSSION THREAD

**Discussion by code-manoj** [aws/aws-sdk-java#2825] [Q&A]
I am working on Eventbridge and publishing events to event-bus using EventBridgeClient.

As show in the below image. I want to override the meta-data fields in the event.

Changes I'm currently trying to implement.
- Covert detail-type field to ( detail-type -> event-type ).
- Convert detail field to ( detail -> payload ).
- And also I want to change the event version from 0 to 0.1

Please let me know. How i can achieve this via aws-sdk code.

![image](https://user-images.githubusercontent.com/111496877/185368964-4cdf7391-becd-4193-b0d7-dc2954be9250.png)

**Comment by debora-ito** [comment]
@code-manoj you can't really change the structure, but you could add custom request fields - just don't know if it would work for what you're trying to do.
